go/vt/vtgate/evalengine/compiler_asm_push.go

```diff
@@ -39,7 +45,11 @@ func (asm *assembler) PushColumn_i(offset int) {
 	asm.adjustStack(1)
 
 	asm.emit(func(env *ExpressionEnv) int {
-		return push_i(env, env.Row[offset].Raw())
+		col := env.Row[offset]
+		if col.IsNull() {
```

Comment:
It was not possible to get a `NULL` here before or was this an existing bug that was uncovered? And could we avoid emitting this if we know it's `NOT NULL`?

Reply:
This is an existing bug! It would crash the compiler when a value from a nullable column contains a `null` (which is, huh, very frequent). The extra null checks can be removed by specializing: we would need to add a dependency on `Fields`, however, which the compiler doesn't have right now. Alternatively, we'd need to gather the nullability information from the VSchema, which may or may not be supported yet. Definitely something to follow up on a separate PR.